cmd/kubectl-gadget/utils/flags.go

```diff
@@ -122,6 +122,12 @@ func AddCommonFlags(command *cobra.Command, params *CommonFlags) {
 			nodeFound := false
 			for _, node := range nodes.Items {
 				if node.Name == params.Node {
+					value, ok := node.Labels["kubernetes.io/os"]
+					if !ok || value != "linux" {
```

Comment:
This can also happen if the user deploys a cluster with all non-linux nodes or if the cluster was freshly deployed and nodes were still being added so I believe this is the problem of the user?

Reply:
I pushed a commit with an alternative day to deal with this: now traces are based on gadget pods instead of nodes, it makes it easier for us to check on which nodes we'll should create the tracers. What do you think about this approach? 

/cc @alban

Reply:
> I pushed a commit with an alternative day to deal with this: now traces are based on gadget pods instead of nodes, it makes it easier for us to check on which nodes we'll should create the tracers. What do you think about this approach?

I think it is better because the logic is at only one place (the daemon set yaml) instead of being copied at different places in the code. Additionally, @eiffel-fl's concern becomes moot.

Reply:
Updated the PR with this approach. Could you please check again?